src/PortingAssistantExtensionClientShared/Utils/AwsUtils.cs

```diff
@@ -136,5 +157,61 @@ TelemetryConfiguration telemetryConfiguration
                 return false;
             }
         }
+
+        public static async Task<AwsCredential> GetAwsCredentialsAsync(bool enabledDefaultCredentials, string profileName)
+        {
+            var chain = new CredentialProfileStoreChain();
+            AWSCredentials awsCredentials;
+
+            if (enabledDefaultCredentials)
+            {
+                awsCredentials = FallbackCredentialsFactory.GetCredentials();
+                if (awsCredentials == null)
+                {
+                    return null;
+                }
+            }
+            else
+            {
+                if (!chain.TryGetAWSCredentials(profileName, out awsCredentials))
+                {
+                    return null;
+                }
+            }
+
+            var immutableCredentials = await awsCredentials.GetCredentialsAsync();
+            AwsCredential awsCredential = new AwsCredential(immutableCredentials.AccessKey, immutableCredentials.SecretKey, immutableCredentials.Token);
+
+            return awsCredential;
+        }
+
+
+        public static async Task<bool> ValidateProfileAsync()
+        {
+            if (UserSettings.Instance.AWSProfileName != null || UserSettings.Instance.EnabledDefaultCredentials)
+            {
+                AwsCredential awsCredentials = await GetAwsCredentialsAsync(UserSettings.Instance.EnabledDefaultCredentials, UserSettings.Instance.AWSProfileName);
+                if (awsCredentials == null)
+                {
+                    await NotificationUtils.ShowInfoBarAsync(PAGlobalService.Instance.AsyncServiceProvider, "AWS Credentials associated with Porting Assistant for .NET may have Expired. Please refresh credentials.");
+                    return false;
+                }
+
+                var AssemblyPath = System.IO.Path.GetDirectoryName(System.Reflection.Assembly.GetExecutingAssembly().Location);
+                var ConfigurationFileName = Environment.GetEnvironmentVariable("ConfigurationJson") ?? Common.Constants.DefaultConfigurationFile;
+                var ConfigurationPath = System.IO.Path.Combine(
+                    AssemblyPath,
+                    Common.Constants.ResourceFolder,
+                    ConfigurationFileName);
+                var TelemetryConfiguration = JsonConvert.DeserializeObject<PortingAssistantIDEConfiguration>(File.ReadAllText(ConfigurationPath)).TelemetryConfiguration;
+
+                if (!await AwsUtils.VerifyUserAsync("", awsCredentials, TelemetryConfiguration))
```

Comment:
nitpicking, but we can consolidate the two checks into this block. so if (awsCredentials == null || !await AwsUtils.VerifyUserAsync)